### PR TITLE
dynamic file reading in metadraw (+enable multifile reading)

### PR DESCRIPTION
**Pull request by rmillikin**

### CODE REVIEW

File: Test/XLTestNGlyco.cs [hunk 1]
```diff
@@ -75,7 +75,7 @@ public static void GlyTest_GlyGetTheoreticalFragments()
             //    }
             //}
 
-            CommonParameters commonParameters = new CommonParameters(deconvolutionMassTolerance: new PpmTolerance(20));
+            CommonParameters commonParameters = new CommonParameters(deconvolutionMassTolerance: new PpmTolerance(20), trimMsMsPeaks: false);
```
Comment (rmillikin) [inline Test/XLTestNGlyco.cs:78]: I added peak filtering for .mgf files to the new mzLib version, which was missing. I edited this unit test to not filter MS2 peaks to keep the results of the test the same as the old version.

File: Test/XLTest.cs [hunk 2]
```diff
@@ -1168,7 +1168,8 @@ public static void TestWriteToPercolator()
                 XlSearchParameters = new XlSearchParameters
                 {
                     WriteOutputForPercolator = true
-                }
+                },
+                CommonParameters = new CommonParameters(trimMsMsPeaks: false)
```
Comment (rmillikin) [inline Test/XLTest.cs:1172]: I added peak filtering for .mgf files to the new mzLib version, which was missing. I edited this unit test to not filter MS2 peaks to keep the results of the test the same as the old version.

File: Test/XLTest.cs [hunk 3]
```diff
@@ -893,7 +893,7 @@ public static void XLSearchWithGeneratedIndices()
         {
             XLSearchTask xlSearchTask = new XLSearchTask
             {
-                CommonParameters = new CommonParameters()
+                CommonParameters = new CommonParameters(trimMsMsPeaks: false)
```
Comment (rmillikin) [inline Test/XLTest.cs:896]: I added peak filtering for .mgf files to the new mzLib version, which was missing. I edited this unit test to not filter MS2 peaks to keep the results of the test the same as the old version.

File: Test/SilacTest.cs [hunk 4]
```diff
@@ -409,13 +409,13 @@ public static void TestSilacTurnover()
...
             output = File.ReadAllLines(TestContext.CurrentContext.TestDirectory + @"/TestSilac/AllQuantifiedPeaks.tsv");
             Assert.AreEqual(output.Length, 4); //header, unlabeled, mixed, labeled
-            Assert.IsTrue(output[1].Contains("\tPEPTK(+8.014)IDEK(+8.014)\t")); //test the original is present
-            Assert.IsTrue(output[1].Contains("\t218750\t")); //test intensity
-            Assert.IsTrue(output[2].Contains("\tPEPTK(+1.994)IDEK(+8.014)\t")); //test human readable label (and lack thereof) is present
-            Assert.IsTrue(output[2].Contains("\t875000\t")); //test intensity
-            Assert.IsTrue(output[3].Contains("\tPEPTK(+1.994)IDEK(+1.994)\t")); //test other label is present
-            Assert.IsTrue(output[3].Contains("\t437500\t")); //test intensity
-            Assert.IsTrue(output[3].Contains("silac\t")); //test human readable labels are present
+            Assert.IsTrue(output[3].Contains("\tPEPTK(+8.014)IDEK(+8.014)\t")); //test the original is present
```
Comment (rmillikin) [inline Test/SilacTest.cs:412]: The new FlashLFQ (in mzLib) version writes non-zero intensity peaks first, so the ordering of the peaks in the written file is different than what this test expected and the unit test had to be edited a bit to expect a different order.

File: Test/XLTest.cs [hunk 5]
```diff
@@ -1184,7 +1185,7 @@ public static void TestWriteToPercolator()
             var results = Path.Combine(outputFolder, @"TestPercolator\XL_Intralinks_Percolator.txt");
             var lines = File.ReadAllLines(results);
             Assert.That(lines[0].Equals("SpecId\tLabel\tScannr\tScore\tdScore\tCharge\tMass\tPPM\tLenShort\tLenLong\tLenSum\tPeptide\tProtein"));
-            Assert.That(lines[1].Equals("T-1-30.61909926666667\t1\t1\t26.06004534434461\t11.026813997502483\t3\t1994.0520231384269\t0.6649793543976755\t7\t9\t16\t-.EKVLTSSAR2--LSQKFPK4.-\t3336842(211)\t3336842(245)"));
+            Assert.That(lines[1].Equals("T-1-30.61909926666667\t1\t1\t26.06004534434461\t21.06004534434461\t3\t1994.0520231384269\t0.6649793543976755\t7\t9\t16\t-.EKVLTSSAR2--LSQKFPK4.-\t3336842(211)\t3336842(245)"));
```
Comment (rmillikin) [inline Test/XLTest.cs:1188]: This was a result of a bugfix that Lei made in mzLib ( https://github.com/smith-chem-wisc/mzLib/commit/c1dfaee4375798302c36be7f2e8bc54b102718a7 )

**Comment by lonelu** [comment]
Thanks for the hard work on dynamic file reading. This is super useful for MetaDraw.

**Pull request merged**